core/common/src/main/java/alluxio/util/LogUtils.java

```diff
@@ -113,4 +113,23 @@ private static void process(java.util.logging.Logger log, String level, LogInfo
     }
     result.setLevel(lev.toString());
   }
+
+  /**
+   * Log a warning message with full exception if debug logging is enabled,
+   * or just the message otherwise.
+   *
+   * @param logger the logger to be used
+   * @param message the message to be logged
+   * @param args the arguments for the message
+   */
+  public static void warnWithException(Logger logger, String message, Object ...args) {
```

Comment:
Would the API be cleaner if we had an explicit `Throwable` parameter?

Reply:
There are two challenges implementing this:
- To have a fixed `Throwable` parameter, it has to be put before the `varargs`. That is different than the `slf4j` convention.
- To pass the `Throwable` parameter to the inner `logger.warn` call, it has to be appended to the `varargs` array. That introduces some perf overhead by calling array copy.
Thoughts?

Reply:
I was thinking to put it before the varargs, but since the ugliness is restricted to this method, it might be more worth to preserve the slf4j convention.